lib/rspec/support/differ.rb

```diff
@@ -77,6 +101,10 @@ def no_procs?(*args)
         safely_flatten(args).none? { |a| Proc === a }
```

Comment:
What I really liked about your previous changes was the potential to get rid of the recursive check and the safely_flatten method.
No pressure, as this is not the goal of the pr, but it would be a nice bonus to fixing the issue. And a performance improvement in theory.

Reply:
I can’t tell of the top of my head with certainty, but I can’t recall if nested hashes are (usefully) diffed.

I’d rather simplify the code if no spec fails.